File: old/src/main/kotlin/UI/BuyerViewModel.cs
using AutoBuyer.Logic;

namespace AutoBuyer.UI
{
    public class BuyerViewModel : ViewModel
    {
        public String ItemId { get; }
        public String CurrentPrice { get; private set; }
        public String NumberInStock { get; private set; }
        public String BoughtSoFar { get; private set; }
        public String State { get; private set; }

        public BuyerViewModel(String itemId, BuyerSnapshot snapshot)
        {
            ItemId = itemId;
            UpdateState(snapshot);
        }

        public void UpdateState(BuyerSnapshot snapshot)
        {
            CurrentPrice = snapshot.CurrentPrice.ToString();
            NumberInStock = snapshot.NumberInStock.ToString();
            BoughtSoFar = snapshot.BoughtSoFar.ToString();
            State = snapshot.State.ToString();

            Notify(nameof(CurrentPrice));
            Notify(nameof(NumberInStock));
            Notify(nameof(BoughtSoFar));
            Notify(nameof(State));
        }
    }
}

File: old/src/main/kotlin/UI/BuyerViewModel.kt
class BuyerViewModel : ViewModel {
    val itemId:String
    var currentPrice:String = ""
        private set
    var numberInStock:String = ""
        private set
    var boughtSoFar:String = ""
        private set
    var state:String = ""
        private set

    constructor(itemId: String, snapshot: BuyerSnapshot) {
        this.itemId = itemId
        updateState(snapshot)
    }

    fun updateState( snapshot:BuyerSnapshot)
    {
        currentPrice = snapshot.currentPrice.toString()
        numberInStock = snapshot.numberInStock.toString()
        boughtSoFar = snapshot.boughtSoFar.toString()
        state = snapshot.state.toString()

        notify(nameof(currentPrice))
        notify(nameof(numberInStock))
        notify(nameof(boughtSoFar))
        notify(nameof(state))
    }
}
